worldedit-core/src/main/java/com/sk89q/worldedit/extent/clipboard/io/legacycompat/BannerBlockCompatibilityHandler.java

```diff
@@ -0,0 +1,171 @@
+        } catch (NullPointerException | IllegalArgumentException | ClassCastException e) {
+            tempFacing = null;
+            tempRotation = null;
+        }
+        FacingProperty = tempFacing;
+        RotationProperty = tempRotation;
+    }
+
+    @Override
+    public <B extends BlockStateHolder<B>> boolean isAffectedBlock(B block) {
+        return block.getBlockType() == BlockTypes.WHITE_BANNER
+                || block.getBlockType() == BlockTypes.WHITE_WALL_BANNER;
+    }
+
+    @Override
+    public <B extends BlockStateHolder<B>> B updateNBT(B block, Map<String, Tag> values) {
+        Tag typeTag = values.get("Base");
+        if (typeTag instanceof IntTag) {
+            boolean isWall = block.getBlockType() == BlockTypes.WHITE_WALL_BANNER;
+            String bannerType = convertBannerType(((IntTag) typeTag).getValue(), isWall);
+            if (bannerType != null) {
+                BlockType type = BlockTypes.get("minecraft:" + bannerType);
+                if (type != null) {
+                    BlockState state = type.getDefaultState();
+
+                    if (isWall) {
+                        Property facingProp = type.getProperty("facing");
+                        state = state.with(facingProp, block.getState(FacingProperty));
+                    } else {
+                        Property rotationProp = type.getProperty("rotation");
+                        state = state.with(rotationProp, block.getState(RotationProperty));
+                    }
+
+                    values.remove("Base");
+
+                    Tag patternsTag = values.get("Patterns");
+                    if (patternsTag instanceof ListTag) {
+                        List<Tag> tempList = new ArrayList<>();
+                        for (Tag pattern : ((ListTag) patternsTag).getValue()) {
+                            if (pattern instanceof CompoundTag) {
+                                Map<String, Tag> patternMap = ((CompoundTag) pattern).getValue();
+                                Tag colorTag = patternMap.get("Color");
+
+                                CompoundTagBuilder builder = CompoundTagBuilder.create();
+                                builder.putAll(patternMap);
+                                if (colorTag instanceof IntTag) {
+                                    builder.putInt("Color", 15 - ((IntTag) colorTag).getValue());
+                                }
+                                tempList.add(builder.build());
+                            } else {
+                                tempList.add(pattern);
+                            }
+                        }
+                        values.put("Patterns", new ListTag(((ListTag) patternsTag).getType(), tempList));
+                    }
+                    return (B) state;
+                }
+            }
+        }
+        return block;
+    }
+
+    private static String convertBannerType(Integer oldType, boolean isWall) {
```

Comment:
This doesn't have to be `Integer` -- the `IntTag` can never return `null`, if you look at it.

Same for beds :)

Reply:
I am updating the PR to reflect this. Is there a reason that `getValue()` doesn't return type `int`. Not that it makes a practical difference

Reply:
Yes, inheritance. Super type returns Object.